### PR TITLE
qnx support

### PR DESCRIPTION
**Pull request by asobhy-qnx**
Add necessary changes to build Lanelet2 for QNX RTOS

### CODE REVIEW

**Comment by asobhy-qnx** [comment]
@m-naumann is it possible to merge this PR

File: lanelet2_core/include/lanelet2_core/Exceptions.h [hunk 1]
```diff
@@ -2,6 +2,10 @@
 #include <stdexcept>
 #include <vector>
 
+#ifdef __QNXNTO__
```
Comment (poggenhans) [inline lanelet2_core/include/lanelet2_core/Exceptions.h:5]: Just unconditionally include `<string>`. It seems like this include is missing anyways.

File: lanelet2_io/src/OsmHandlerLoad.cpp [hunk 2]
```diff
@@ -1,3 +1,8 @@
+
+#ifdef __QNXNTO__
```
Comment (poggenhans) [inline lanelet2_io/src/OsmHandlerLoad.cpp:2]: here too

File: lanelet2_io/src/OsmHandlerLoad.cpp [hunk 3]
```diff
@@ -408,7 +413,12 @@ void registerIds(const MapT& map) {
 }
 
 void testAndPrintLocaleWarning(ErrorMessages& errors) {
+  struct lconv * lc = localeconv();
```
Comment (poggenhans) [inline lanelet2_io/src/OsmHandlerLoad.cpp:416]: I think you can get rid of this change by just unconditionally including `<clocale>`, this include seems to be missing too.

File: lanelet2_io/src/OsmHandlerWrite.cpp [hunk 4]
```diff
@@ -6,6 +6,10 @@
 #include "lanelet2_io/io_handlers/OsmFile.h"
 #include "lanelet2_io/io_handlers/OsmHandler.h"
 
+#ifdef __QNXNTO__
```
Comment (poggenhans) [inline lanelet2_io/src/OsmHandlerWrite.cpp:9]: ...and here too. Just including `<clocale>` in all cases should do it.

**Review by poggenhans — COMMENTED** [review]
Thanks for this PR, there is no reason why we shouldn't support qnx.

But I think your changes are a bit too intrusive. I added a few comments on how to solve this better. And please don't forget to run `clang-format` on the files, otherwise our CI will reject your PR.

**Comment by asobhy-qnx** [comment]
Thank you for the review. Changes requested have been applied

**Comment by asobhy-qnx** [comment]
hi @poggenhans,

Can you please merge this?

Thanks,
Ahmed

**Pull request closed without merging**